packages/sdk/src/cosmwasmclient.ts

```diff
@@ -166,6 +190,42 @@ export class CosmWasmClient {
     };
   }
 
+  public async getCodes(): Promise<readonly Code[]> {
+    const result = await this.restClient.listCodeInfo();
+    return result.map(r => ({
+      id: r.id,
+      creator: r.creator,
+      checksum: Encoding.toHex(Encoding.fromHex(r.code_hash)),
+      source: r.source || undefined,
+      builder: r.builder || undefined,
+    }));
+  }
+
+  public async getCodeDetails(codeId: number): Promise<CodeDetails> {
+    const result = await this.restClient.getCode(codeId);
+    return {
+      wasm: result,
+    };
+  }
+
+  public async getContracts(codeId: number): Promise<readonly Contract[]> {
```

Comment:
Fine for current rest api, but we do want to add the contract address to all of these soon. I guess it is blocked on wasmd changes, but very important for usa ility.

For the name app, I added address to the api, pulling the argument from get contact and just brute force downloading all contracts and filtering locally for get contracts. This gives me the proper api now, and I can clean up the implementation once wasmd is enhanced.

Reply:
Sounds like a good approach and I agree we want the address here ASAP (make a comment in the `Contract` interface already). Can you show me that code? I don't know how address generation works for contract instances and find it hard to follow your explanation.

Reply:
I found the code/implementation in name-app you're refering to. This is .... wow, creative. And it does the job 🎩. But even in our small short living testing world this would easily send 100 requests to get the addresses of a single `codeId`. If there is no very good reason we need the interface now, I'd wait until the backend is ready.